selectors/selectors.md

```diff
@@ -108,3 +108,269 @@ These components imply or expand into the following things:
 - Allow language-specific implementations of selectors (parsers, execution, etc).
 - (IMPORTANT) Well-designed set of test vectors representing a variety of use cases for IPLD Selectors.
 - A recommended structure for implementing a selector type, with an easy to use test suite.
+
+
+Selector types
+--------------
+
+### CID Selector
+
+Selects a single CID.
+
+
+### Path Selector
+
+Selects a single path segment.
+
+
+### Array Selector
+
+Selects a slice or single item of an array of items.
+
+
+### Recursive Selector
```

Comment:
Let's say that I want to lookup a specific path on each block along a chain. I think we'll need a way to fork. Maybe something like:

```
{
  "recursive": {
    "follow": ...,
    "select": ...,
    "stop": ...,
  }
}
```

Reply:
In the current spec we don't have any filtering capability at all yet. Currently we just return everythig we traverse. I'll open an issue for that once this spec is merged.

Reply:
https://github.com/ipld/specs/issues/106